### ISSUE TITLE
Call graph enabled crashes server when using y_hooks

### ISSUE THREAD

**Issue by Misiur** [Zeex/samp-plugin-profiler#18]
# Summary

When using y_hooks, enabling call graph will crash the server
# Server log

```
----------
Loaded log file: "server_log.txt".
----------

SA-MP Dedicated Server
----------------------
v0.3.7, (C)2005-2015 SA-MP Team

[14:32:12] 
[14:32:12] Server Plugins
[14:32:12] --------------
[14:32:12]  Loading plugin: streamer
[14:32:12] 

*** Streamer Plugin v2.6.1 by Incognito loaded ***

[14:32:12]   Loaded.
[14:32:12]  Loading plugin: sscanf
[14:32:12] 

[14:32:12]  ===============================

[14:32:12]       sscanf plugin loaded.     

[14:32:12]    (c) 2009 Alex "Y_Less" Cole

[14:32:12]    0.3d-R2 500 Players "dnee"

[14:32:12]  ===============================

[14:32:12]   Loaded.
[14:32:12]  Loading plugin: audio
[14:32:12] 

*** Audio Plugin v0.5 R2 by Incognito loaded ***

[14:32:12]   Loaded.
[14:32:12]  Loading plugin: Whirlpool
[14:32:12]  
[14:32:12]  ==================
[14:32:12]  
[14:32:12]   Whirlpool loaded
[14:32:12]  
[14:32:12]  ==================
[14:32:12]  
[14:32:12]   Loaded.
[14:32:12]  Loading plugin: mysql
[14:32:12]  >> plugin.mysql: R39-3 successfully loaded.
[14:32:12]   Loaded.
[14:32:12]  Loading plugin: crashdetect
[14:32:12]   CrashDetect v4.15.1 is OK.
[14:32:12]   Loaded.
[14:32:12]  Loading plugin: profiler
[14:32:12]   Profiler v2.12.1 is OK.
[14:32:12]   Loaded.
[14:32:12]  Loaded 7 plugins.

[14:32:12] 
[14:32:12] Filterscripts
[14:32:12] ---------------
[14:32:12]   Loaded 0 filterscripts.

[14:32:13] [profiler] Attached profiler to gamemodes/ugrp.amx
[14:32:13] Hooks_OnScriptInit called
[14:32:13] AMX_GetPointerPrefix called: 0, 0, 0, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 0, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetPointerPrefix called: 0, 694, 187584, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 694, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetPointerPrefix called: 0, 695, 188004, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 695, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetPointerPrefix called: 0, 696, 187644, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 696, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetPointerPrefix called: 0, 697, 187764, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 697, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetPointerPrefix called: 0, 698, 187704, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 698, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetPointerPrefix called: 0, 699, 187944, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 699, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetPointerPrefix called: 0, 700, 187824, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 700, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetPointerPrefix called: 0, 701, 187884, 1215913792
[14:32:13] AMX_GetEntryPrefix called: 0, 701, 0, 1215913792
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetEntryPrefix called: 0, 0, 0, 1599030080
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetEntryPrefix called: 0, 0, -8, 1599030080
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetEntryPrefix called: 0, 2, -3770288, 1599030080
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetEntryPrefix called: 0, 3, -3770280, 1599030080
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_GetEntryPrefix called: 0, 4, -3770272, 1599030080
[14:32:13] AMX_GetBaseCount called: 0, 0, 0
[14:32:13] AMX_Read called: -3770288 -152
[14:32:13] AMX_RawRead called: -3770288
[14:32:13] AMX_Read called: -3770284 -160
[14:32:13] AMX_RawRead called: -3770284
[14:32:13] AMX_Read called: -3770292 -12
[14:32:13] AMX_RawRead called: -3770292
[14:32:13] AMX_Read called: -3770284 -16
[14:32:13] AMX_RawRead called: -3770284
[14:32:13] AMX_Read called: -3754022 -8
[14:32:13] AMX_RawRead called: -3754022
[14:32:13] AMX_Read called: -3754019 -12
[14:32:13] AMX_RawRead called: -3754019
[14:32:13] [debug] Server crashed while executing ugrp.amx
[14:32:13] [debug] AMX backtrace:
[14:32:13] [debug] #0 0002eaf8 in Hooks_GetPreloadLibraries (preloads[][E_PRE_HOOK:17]=@03c39124, &precount=@03c39364 3, size=7) at C:\Work\ugrp\pawno\include\YSI\..\YSI_Coding\y_hooks/impl.inc:320
[14:32:13] [debug] #1 0002f9f4 in Hooks_DoAllHooks () at C:\Work\ugrp\pawno\include\YSI\..\YSI_Coding\y_hooks/impl.inc:589
[14:32:13] [debug] #2 000318a0 in public AMX_OnScriptInit () at C:\Work\ugrp\pawno\include\YSI\..\YSI_Coding\y_hooks/impl.inc:1144
[14:32:13] [debug] #3 00008934 in public Debug_OnScriptInit () at C:\Work\ugrp\pawno\include\YSI\..\YSI_Coding\..\YSI_Core\..\YSI_Storage\y_amx.inc:209
[14:32:13] [debug] #4 00001db8 in public YVers_OnScriptInit () at C:\Work\ugrp\pawno\include\YSI\..\YSI_Coding\..\YSI_Core\..\YSI_Core\y_debug.inc:340
[14:32:13] [debug] #5 0000183c in public ScriptInit_OnScriptInit () at C:\Work\ugrp\pawno\include\YSI\..\YSI_Coding\..\YSI_Core\..\YSI_Internal\y_version.inc:140
[14:32:13] [debug] #6 00001640 in public OnGameModeInit () at C:\Work\ugrp\pawno\include\YSI\..\YSI_Coding\..\YSI_Core\..\YSI_Internal\..\YSI_Server\y_scriptinit.inc:183
[14:32:13] [debug] Native backtrace:
[14:32:13] [debug] #0 66d687d2 in ?? () from plugins\profiler.DLL
[14:32:13] [debug] #1 66d68e9f in ?? () from plugins\profiler.DLL
[14:32:13] [debug] #2 66d60fba in ?? () from plugins\profiler.DLL
[14:32:13] [debug] #3 66ea8c75 in ?? () from plugins\crashdetect.DLL
[14:32:13] [debug] #4 66ea09c7 in ?? () from plugins\crashdetect.DLL
[14:32:13] [debug] #5 66ea631a in ?? () from plugins\crashdetect.DLL
[14:32:13] [debug] #6 66d6903b in ?? () from plugins\profiler.DLL
[14:32:13] [debug] #7 66d6119e in ?? () from plugins\profiler.DLL
[14:32:13] [debug] #8 66d643ba in ?? () from plugins\profiler.DLL
[14:32:13] [debug] #9 0046f504 in ?? () from samp-server.exe
[14:32:13] [debug] #10 65646f6d in ?? () from samp-server.exe
[14:32:13] [debug] #11 67752f73 in ?? () from C:\Windows\system32\MSVCP120.dll
[14:32:13] [debug] #12 612e7072 in ?? () from samp-server.exe
```
# Steps to reproduce
1. Create bare gamemode file
2. Include y_hooks
3. Enable call graph
# Environment
1. Windows 7 x64 Pro with Emet 5 enabled
2. Samp-server 0.3.7
3. Profiler 2.12.1 (2.12 crashes as well, so it's not a regression since last crash fix)

**Comment by rt-2** [comment]
This is what I get:
`
----------
Loaded log file: "server_log.txt".
----------

SA-MP Dedicated Server
----------------------
v0.3.7-R2, (C)2005-2015 SA-MP Team

[12/06/16 21:49:55] 
[12/06/16 21:49:55] Server Plugins
[12/06/16 21:49:55] --------------
[12/06/16 21:49:55]  Loading plugin: crashdetect
[12/06/16 21:49:55]   CrashDetect v4.17 is OK.
[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: profiler
[12/06/16 21:49:55]   Profiler v2.14 is OK.
[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: SKY
[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: YSF
[12/06/16 21:49:55] logprintf = 0x0048C8D0

[12/06/16 21:49:55] 

[12/06/16 21:49:55]  ===============================

[12/06/16 21:49:55]         YSF - kurta999 version R16 loaded

[12/06/16 21:49:55]    (c) 2008 Alex "Y_Less" Cole - (c) 2010 - 2015 kurta999

[12/06/16 21:49:55]     Server version: 0.3.7 R2-1

[12/06/16 21:49:55]     Operating System: Windows

[12/06/16 21:49:55]     Built on: Feb  2 2016 at 22:53:45

[12/06/16 21:49:55]  ===============================

[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: mysql
[12/06/16 21:49:55]  >> plugin.mysql: R39-6 successfully loaded.
[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: streamer
[12/06/16 21:49:55] 

*** Streamer Plugin v2.7.9 by Incognito loaded ***

[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: FileFunctions
[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: FCNPC
[12/06/16 21:49:55] 
[12/06/16 21:49:55] -------------------------------------------------
[12/06/16 21:49:55]      FCNPC - Fully Controllable NPC v1.0.5
[12/06/16 21:49:55] 
[12/06/16 21:49:55] - Author: OrMisicL
[12/06/16 21:49:55] - Contributors: ziggi, Neutralneu
[12/06/16 21:49:55] - Server version: 0.3.7 R2-1
[12/06/16 21:49:55] - Build: Apr 17 2016 at 01:34:41
[12/06/16 21:49:55] -------------------------------------------------
[12/06/16 21:49:55] 
[12/06/16 21:49:55] Loading...
[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: PathFinder
[12/06/16 21:49:55] =========================================
[12/06/16 21:49:55] PathFinder Plugin 1.0 MT
[12/06/16 21:49:55]          by Pamdex
[12/06/16 21:49:55] 
[12/06/16 21:49:55] Using MapAndreas Plugin 1.2.1
[12/06/16 21:49:55] Waiting for Init...
[12/06/16 21:49:55] =========================================
[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loading plugin: MapAndreas
[12/06/16 21:49:55]   Loaded.
[12/06/16 21:49:55]  Loaded 10 plugins.

[12/06/16 21:49:55] 
[12/06/16 21:49:55] Ban list
[12/06/16 21:49:55] --------
[12/06/16 21:49:55]  Loaded: samp.ban
[12/06/16 21:49:55] 
[12/06/16 21:49:55] 
[12/06/16 21:49:55] Filterscripts
[12/06/16 21:49:55] ---------------
[12/06/16 21:49:55]   Loading filterscript 'antiddos.amx'...
[12/06/16 21:50:01] [profiler] Attached profiler to filterscripts/antiddos (no debug info)
[12/06/16 21:50:01] [profiler] Started profiling filterscripts/antiddos
[12/06/16 21:50:01]  
[12/06/16 21:50:01]  
[12/06/16 21:50:01]  
[12/06/16 21:50:01]  ======================================= 
[12/06/16 21:50:01]  |                                     | 
[12/06/16 21:50:01]  |        YSI version 4.00.0001        | 
[12/06/16 21:50:01]  |        By Alex "Y_Less" Cole        | 
[12/06/16 21:50:01]  |                                     | 
[12/06/16 21:50:01]  ======================================= 
[12/06/16 21:50:01]  `

When I remove 
`profiler_callgraph 1
profiler_callgraphformat dot`
from my server.cfg, it works.

Thank you

**Comment by Zeex** [comment]
Fixed in 2.15.3
